### PR TITLE
[FLINK-7617] Remove string format in BitSet to improve the performance of BuildSideOuterjoin

### PR DESCRIPTION
**Pull request by JingsongLi**
## What is the purpose of the change
Remove string format in BitSet to improve the performance of BuildSideOuterjoin

## Brief change log
  - BitSet.set
  - BitSet.get

## Documentation
  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by JingsongLi** [comment]
## Performance
  - HashVsSortMiniBenchmark: use buildSideOuterJoin
testBuildFirst:       16s 208ms(before)    10s 98ms(after)
testBuildSecond:    7s 793ms(before)     6s 67ms(after)

R: @KurtYoung

**Comment by KurtYoung** [comment]
Thanks @JingsongLi for your contribution
LGTM, +1 to merge
cc @fhueske @StephanEwen

File: flink-runtime/src/main/java/org/apache/flink/runtime/operators/util/BitSet.java [hunk 1]
```diff
@@ -56,8 +56,7 @@ public void setMemorySegment(MemorySegment memorySegment, int offset) {
 	 * @param index - position
 	 */
 	public void set(int index) {
-		Preconditions.checkArgument(index < bitLength && index >= 0, 
-			String.format("Input Index[%d] is larger than BitSet available size[%d].", index, bitLength));
+		Preconditions.checkArgument(index < bitLength && index >= 0);
```
Comment (fhueske) [inline flink-runtime/src/main/java/org/apache/flink/runtime/operators/util/BitSet.java]: Can be changed to 

```
Preconditions.checkArgument(index < bitLength && index >= 0, 
  "Input Index[%d] is larger than BitSet available size[%d].", index, bitLength);
``` 
to perserve the error message.

Comment (fhueske) [reply flink-runtime/src/main/java/org/apache/flink/runtime/operators/util/BitSet.java]: Just noticed your reply to the same comment on the JIRA issue. You are of course right, the varargs will create and fill an array which adds overhead.

So +1 to keep it as it is.

**Comment by fhueske** [comment]
+1 to merge this

**Comment by fhueske** [comment]
Hi @KurtYoung, do you want to merge this to the `release-1.3` branch as well?

**Comment by KurtYoung** [comment]
@fhueske sure

**Pull request closed without merging**